### ISSUE TITLE
Improve UI/UX about persistent storage

### ISSUE THREAD

**Issue by maskedriotuser** [vector-im/riot-web#14418]
Hello,

When connecting to https://riot.im/app/ for the first time on a new device, Firefox immediately displays a dialog box to allow data storage:

![datastorage](https://user-images.githubusercontent.com/68146982/87223445-84caf800-c36c-11ea-9d77-ef12231ec63b.png)

However, at this time, the user might be completely new to Riot, and not even know if he really wants to use it or not.

But  it's confusing even for people who already are users. What data will be stored? Is it absolutely necessary? Is a functionality of Riot not going to work if the user clicks on "Not now", or is it just for convenience? Will the website prompt later for consent if needed, or skipping consent now risk breaking a functionality later?

In conclusion, It could be better to initiate data storage only after having informed the user of what data is going to be stored, and why. For example on the pages after having clicked on "Sign in" or "Create Account", which could display a simple phrase explaining the purpose of data storage.

**Comment by t3chguy** [comment]
> What data will be stored?

Given Riot runs entirely in your browser, everything it needs;
+ Your session
+ Your encryption keys
+ The latest chunks of timeline

>  Is it absolutely necessary?

Nope but without it Firefox can purge this data whenever it deems it wants to, especially when you are running low on space.

> Is a functionality of Riot not going to work if the user clicks on "Not now", or is it just for convenience?

It is for peace of mind that your data which you may not have a backup of does not get wiped without your explicit act.

> It could be better to initiate data storage only after having informed the user of what data is going to be stored, and why. 

This isn't possible as Firefox in many configurations lies on its UA saying it is not firefox/similar and this prompt only exists in Firefox, in Chrome it is much more automatic.
